packages/client/modules/userDashboard/components/OrgBilling/NewBillingLeaderMenu.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  NewBillingLeaderMenu_organization$data,
+  NewBillingLeaderMenu_organization$key
+} from '~/__generated__/NewBillingLeaderMenu_organization.graphql'
+import Avatar from '../../../../components/Avatar/Avatar'
+import styled from '@emotion/styled'
+import TypeAheadLabel from '../../../../components/TypeAheadLabel'
+import useFilteredItems from '../../../../hooks/useFilteredItems'
+import {EmptyDropdownMenuItemLabel} from '../../../../components/EmptyDropdownMenuItemLabel'
+import SetOrgUserRoleMutation from '../../../../mutations/SetOrgUserRoleMutation'
+import useAtmosphere from '../../../../hooks/useAtmosphere'
+import useMutationProps from '../../../../hooks/useMutationProps'
+
+const AvatarBlock = styled('div')({
+  paddingRight: 32
+})
+
+interface Props {
+  menuProps: MenuProps
+  organizationRef: NewBillingLeaderMenu_organization$key
+  newLeaderSearchQuery: string
+}
+
+const getValue = (
+  orgUser: NewBillingLeaderMenu_organization$data['organizationUsers']['edges'][0]
+) => orgUser.node.user.preferredName
+
+const NewBillingLeaderMenu = forwardRef((props: Props, ref: any) => {
+  const {menuProps, organizationRef, newLeaderSearchQuery} = props
+  const atmosphere = useAtmosphere()
+  const {onError, onCompleted} = useMutationProps()
+  const organization = useFragment(
+    graphql`
+      fragment NewBillingLeaderMenu_organization on Organization {
+        id
+        billingLeaders {
+          id
+        }
+        organizationUsers {
+          edges {
+            node {
+              id
+              user {
+                id
+                preferredName
+                picture
+              }
+            }
+          }
+        }
+      }
+    `,
+    organizationRef
+  )
+  const {id: orgId, organizationUsers, billingLeaders} = organization
+  const nonLeaderOrgUsers = useMemo(() => {
+    return organizationUsers.edges.filter((organizationUser) => {
+      const {node} = organizationUser
+      const {user} = node
+      const {id: userId} = user
+      return !billingLeaders.some((billingLeader) => {
+        const {id: billingLeaderId} = billingLeader
+        return billingLeaderId === userId
+      })
+    })
+  }, [billingLeaders, organizationUsers])
+
+  const query = newLeaderSearchQuery.toLowerCase()
+  const filteredOrgUsers = useFilteredItems(query, nonLeaderOrgUsers, (orgUser) =>
+    getValue(orgUser).toLowerCase()
+  )
+
+  const handleClick = (userId: string) => {
+    const role = 'BILLING_LEADER'
+    const variables = {orgId, userId, role}
+    SetOrgUserRoleMutation(atmosphere, variables, {onError, onCompleted})
+  }
+
+  return (
+    <Menu ariaLabel='Select New Billing Leader' keepParentFocus {...menuProps}>
+      {filteredOrgUsers.length === 0 && (
+        <EmptyDropdownMenuItemLabel key='no-results'>
+          No team members found!
+        </EmptyDropdownMenuItemLabel>
+      )}
+      {filteredOrgUsers.slice(0, 10).map((organizationUser) => {
```

Comment:
Is it possible to make scroll?

Reply:
We have five other menus that follow this pattern, and the user can always search if they don't see the item they want, so IMO it's safe to keep it like this